modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/impl/TestMvPartitionStorage.java

```diff
@@ -71,7 +72,10 @@ public class TestMvPartitionStorage implements MvPartitionStorage {
 
     private volatile boolean rebalance;
 
-    final LockByRowId lockByRowId = new LockByRowId();
+    private final LockByRowId lockByRowId = new LockByRowId();
+
+    /** Amount of cursors that opened and still do not close. */
+    private final AtomicInteger pendingCursors = new AtomicInteger();
```

Comment:
It's also required to check that index cursors and index lookup cursors are closed, see 

- PartitionReplicaListener#scanSortedIndex()
- PartitionReplicaListener#lookupIndex()

It also means that if we will populate PartitionReplicaListener#cursors with any new cursors e.g. TxStateStorage#scan given test won't catch that. By the way seems that we've missed TxStateStorage#scan closing.